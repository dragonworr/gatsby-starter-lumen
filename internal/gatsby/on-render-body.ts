import { RenderBodyArgs } from "gatsby";

const onRenderBody = ({ setHtmlAttributes }: RenderBodyArgs) => {
  setHtmlAttributes({ lang: "en" });
};

export { onRenderBody };
