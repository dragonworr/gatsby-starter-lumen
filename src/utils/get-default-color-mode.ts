const getDefaultColorMode = (): "dark" | "light" => {
  const mql = window.matchMedia("(prefers-color-scheme: dark)");
  const hasMediaQueryPreference = typeof mql.matches !== "undefined";
  if (hasMediaQueryPreference) {
    return mql.matches ? "dark" : "light";
  }

  return "light";
};

export default getDefaultColorMode;
