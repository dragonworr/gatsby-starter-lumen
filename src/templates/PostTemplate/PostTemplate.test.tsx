import React from "react";
import renderer from "react-test-renderer";

import { render as reactTestingLibraryRender } from "@testing-library/react";
import { StaticQuery, useStaticQuery } from "gatsby";

import * as mocks from "@/mocks";
import { getMeta } from "@/utils";

import PostTemplate, { Head as GatsbyHead } from "./PostTemplate";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("PostTemplate", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  test("renders correctly", () => {
    const props = {
      data: {
        markdownRemark: mocks.markdownRemark,
      },
    };

    const tree = renderer.create(<PostTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("head renders correctly", () => {
    const props = {
      data: {
        markdownRemark: mocks.markdownRemarkWithoutDescription,
      },
    };

    reactTestingLibraryRender(<GatsbyHead {...props} />);

    expect(getMeta("twitter:card")).toEqual("summary_large_image");
    expect(getMeta("twitter:title")).toEqual(
      "Humane Typography in the Digital Age - Blog by John Doe",
    );
    expect(getMeta("og:title")).toEqual(
      "Humane Typography in the Digital Age - Blog by John Doe",
    );
    expect(getMeta("description")).toEqual(
      "Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.",
    );
    expect(getMeta("twitter:description")).toEqual(
      "Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.",
    );
    expect(getMeta("og:description")).toEqual(
      "Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.",
    );
  });
});
