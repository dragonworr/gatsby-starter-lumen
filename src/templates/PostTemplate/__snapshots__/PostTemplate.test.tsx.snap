// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTemplate renders correctly 1`] = `
<div>
  <div>
    <div>
      <a
        className=""
        href="/"
      >
        All Articles
      </a>
      <div
        className=""
      >
        <button
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="24"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="24"
          >
            <title>
              light
            </title>
            <path
              d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
              pathLength="1"
            />
            <circle
              cx="12"
              cy="12"
              pathLength="1"
              r="5"
            />
            <line
              pathLength="1"
              x1="12"
              x2="12"
              y1="1"
              y2="3"
            />
            <line
              pathLength="1"
              x1="12"
              x2="12"
              y1="21"
              y2="23"
            />
            <line
              pathLength="1"
              x1="4.22"
              x2="5.64"
              y1="4.22"
              y2="5.64"
            />
            <line
              pathLength="1"
              x1="18.36"
              x2="19.78"
              y1="18.36"
              y2="19.78"
            />
            <line
              pathLength="1"
              x1="1"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              pathLength="1"
              x1="21"
              x2="23"
              y1="12"
              y2="12"
            />
            <line
              pathLength="1"
              x1="4.22"
              x2="5.64"
              y1="19.78"
              y2="18.36"
            />
            <line
              pathLength="1"
              x1="18.36"
              x2="19.78"
              y1="5.64"
              y2="4.22"
            />
          </svg>
        </button>
      </div>
    </div>
    <div>
      <div>
        <h1>
          Perfecting the Art of Perfection
        </h1>
        <div
          dangerouslySetInnerHTML={
            {
              "__html": "<p>An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.</p>",
            }
          }
        />
      </div>
    </div>
    <div>
      <div>
        <p>
          Published
           
          Sep 1, 2016
        </p>
      </div>
      <div>
        <p>
          Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.
          <a
            href="https://www.twitter.com/#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <strong>
              John Doe
            </strong>
             on Twitter
          </a>
        </p>
      </div>
    </div>
    <div />
  </div>
</div>
`;
